### PR TITLE
Fix reachability analysis for ?? operator

### LINKED ISSUE #34635: Code is incorrectly marked as unreachable when using nullish coalescing
**TypeScript Version:**  3.7.0-beta

**Search Terms:**

Unreachable code detected

**Code**

```ts
const maybeBool = false;
if (!(maybeBool ?? true)) {
    console.log('...'); // <-- marked as unreachable
}
```

**Playground Link:** 

https://www.typescriptlang.org/play/index.html?ts=3.7-Beta&ssl=5&ssc=1&pln=1&pc=1#code/MYewdgzgLgBAtgQwJ4CMCmAhEIA2MC8MAZgjhGgNwBQAlkTABQCEDiqm2eA-FzFAE4BXNAEoRMAN5UYMmKEi40AOhwgA5gwDkSnZpHUAvlSA

### PR DESCRIPTION
**Pull request by ahejlsberg**
Fixes #34635.

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
@typescript-bot cherry-pick this to release-3.7

File: tests/baselines/reference/api/typescript.d.ts [hunk 1]
```diff
@@ -3525,6 +3525,7 @@ declare namespace ts {
     function isCallExpression(node: Node): node is CallExpression;
     function isCallChain(node: Node): node is CallChain;
     function isOptionalChain(node: Node): node is PropertyAccessChain | ElementAccessChain | CallChain;
+    function isQuestionQuestionExpression(node: Node): boolean;
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/api/typescript.d.ts]: Why not

`isNullishCoalesceExpression(node: Node): node is NullCoalesceExpression`

File: src/compiler/utilities.ts [hunk 2]
```diff
@@ -5953,6 +5953,10 @@ namespace ts {
         return isOptionalChainRoot(node.parent) && node.parent.expression === node;
     }
 
+    export function isQuestionQuestionExpression(node: Node) {
```
Comment (DanielRosenwasser) [inline src/compiler/utilities.ts]: The analogous factory function is `createNullishCoalesce`, so I think this should be `isNullishCoalesce`.

Comment (ahejlsberg) [reply src/compiler/utilities.ts]: Yeah, I guess, though for the uninitiated, nullish coalesce is meaningless and question question is self-explanatory. But hey, gotta have something for the quizzes. 😏

Comment (DanielRosenwasser) [reply src/compiler/utilities.ts]: How else do you expect me to rig TSConf trivia?

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick this to release-3.7

**Pull request merged**